File: Robot2016/Robot2016/Components/Arm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WPILib;
using CSharpRoboticsLib.ControlSystems;

namespace Robot2016.Components
{
    enum armlocation
    {
        Low,
        High,
        Intake,
        Load,
        Lower
    }

    class Arm : MotionControlledSystem
    {
        private SimplePID armPID;
        private Talon armMotor;
        private AnalogPotentiometer m_armPotentiometer;
        /// <summary>
        /// Constructor for the Arm. Doesn't take in any values and initializes the Arm components correctly
        /// </summary>
        public Arm() {
            armMotor = new Talon(1);
            m_armPotentiometer = new AnalogPotentiometer(1);
            armPID = new SimplePID(1, 1, 1);
            base.Controller = armPID;
            base.Motor = armMotor;
            base.Sensor = m_armPotentiometer;
        }
        /// <summary>
        /// Moves the Arm to a new location, which is specified by the input of the function.
        /// </summary>
        /// <param name="armPlace"></param>
        public void newPosition(armlocation armPlace)
        {
            switch (armPlace)
            {
                case armlocation.High:
                    base.SetPoint = 0/*Arbitrary Number*/;

                    break;

                case armlocation.Load:
                    base.SetPoint = 0/*Arbitrary Number*/;

                    break;
                case armlocation.Low:
                    base.SetPoint = 0/*Arbitrary Number*/;

                    break;
                case armlocation.Intake:
                    base.SetPoint = 0/*Arbitrary Number*/;

                    break;
                case armlocation.Lower:
                    base.SetPoint = 0/*Arbitrary Number*/;

                    break;
            }
        }

}
}

File: Robot2016/src/org/usfirst/frc/team4488/robot/Arm.java


package org.usfirst.frc.team4488.robot;
import edu.wpi.first.wpilibj.*;
import JavaRoboticsLib.*;
import JavaRoboticsLib.ControlSystems.MotionControlledSystem;
import JavaRoboticsLib.ControlSystems.SimplePID;
enum armlocation
{
    Low,
    High,
    Intake,
    Load,
    Lower
}

class Arm extends MotionControlledSystem
{
    private SimplePID armPID;
    private Talon armMotor;
    private AnalogPotentiometer m_armPotentiometer;
    
    public Arm() throws Exception {
        armMotor = new Talon(RobotMap.ArmMotor_Channel);
        m_armPotentiometer = new AnalogPotentiometer(RobotMap.ArmPotentiometer);
        armPID = new SimplePID(1, 1, 1);
        super.Controller = armPID;
        super.Motor = armMotor;
        super.Sensor = m_armPotentiometer;
    }
    
    /*
     * This moves the arm to the position requested as the input of the program.
     * Should work, (should)....................................................
     */
    public void newPosition(armlocation armPlace)
    {
        switch (armPlace)
        {
        case High:
            super.setSetPoint(0);

            break;

        case Load:
            super.setSetPoint(0);

            break;
        case Low:
            super.setSetPoint(0);

            break;
        case Intake:
            super.setSetPoint(0);

            break;
        case Lower:
            super.setSetPoint(0);

                break;
        }
    }

}
